src/Homeworks/homework_6/avl_tree.py

```diff
@@ -0,0 +1,260 @@
+from dataclasses import dataclass
+from typing import TypeVar, Generic, Optional, Callable, Iterable
+
+Value = TypeVar("Value")
+
+
+@dataclass
+class TreeNode(Generic[Value]):
+    key: int
+    value: Value
+    height: int = 0
+    left: Optional["TreeNode[Value]"] = None
+    right: Optional["TreeNode[Value]"] = None
+
+
+@dataclass
+class Tree(Generic[Value]):
+    root: Optional[TreeNode[Value]] = None
+    size: int = 0
+
+
+def _get_height(node: TreeNode[Value]) -> int:
+    if node is None:
+        return -1
+    return node.height
+
+
+def _get_balance_factor(node: TreeNode[Value]) -> int:
+    if node is None:
+        return 0
+    return _get_height(node.left) - _get_height(node.right)
+
+
+def _update_height(node: TreeNode[Value]) -> None:
+    node.height = max(_get_height(node.left), _get_height(node.right)) + 1
+
+
+def create_tree() -> Tree[Value]:
+    return Tree()
+
+
+def put(tree: Tree[Value], key: int, value: Value) -> None:
+    new_tree_node = TreeNode(key, value)
+    if tree.root is None:
+        tree.root = new_tree_node
+        tree.size = 1
+        return
+
+    def put_recursion(curr_node: TreeNode[Value]) -> TreeNode[Value]:
+        if curr_node is None:
+            return new_tree_node
+        elif curr_node.key == key:
+            curr_node.value = value
+            tree.size -= 1
+            return curr_node
```

Comment:
Но ведь ты не перенес строку, вообще странное в итоге решение
```suggestion
        if curr_node is None:
            tree.size += 1
            return new_tree_node
        elif curr_node.key == key:
            curr_node.value = value
            return curr_node
```

А строку в конце функции убираешь, теперь +1 будет делаться только если новая вершина создается (сюда по-хорошему еще и создание вершины надо перенести)